Use new versioning script from kubecf-tools

jandubois opened the pull request:
Adds kubecf-tools repo as a submodule per discussion with @viovanov 

The new versioning script was developed via #1013

Fixes #766

Example output:

```
$ make version
0.2.0-1455.gfa0a32f0-dirty
```

Requesting review from @viccuad to make sure it is compatible with catapult and CI.

viccuad commented:
We should update the release docs (https://github.com/cloudfoundry-incubator/kubecf/blob/master/doc/release/release-guidelines.md#files-to-change) with this PR, although it could also go into https://github.com/cloudfoundry-incubator/kubecf/pull/1162

viccuad commented:
The `make lint` job rightfully fails, because there aren't any tags present in the git checkout of the CI…

The git checkout performed by the concourse PR resource doesn't have any tags, nor remotes. We could add the following clutch or something similar to create a tag on the lint and build PR CI jobs:
```
{{- if ( eq $branch "pr" )}}
git tag 0.0.0-$(cat .git/resource/pr)-$(cat .git/resource/head_sha)
{{- end }}
```
This would make lint and build pass on PRs, and leaves the master ones untouched and working, since those use a different concourse resource.

Looking forward to move to Github actions.

jandubois commented:
> The `make lint` job rightfully fails, because there aren't any tags present in the git checkout of the CI…

I wish I had seen this before bothering Mark to help me figure this out...

Since we already use a fork of the github-pr-resource, I've fixed this by fetching tags as well: https://github.com/SUSE/github-pr-resource/commit/e92ede5

I've built a new image locally and updated our pipeline to use that: https://github.com/cloudfoundry-incubator/kubecf/pull/1168

Updated pipeline has been deployed, and it fixed this issue, which is already closed.

viccuad commented:
> We should update the release docs (https://github.com/cloudfoundry-incubator/kubecf/blob/master/doc/release/release-guidelines.md#files-to-change) with this PR, although it could also go into #1162

Updated release-guidelines.md in https://github.com/cloudfoundry-incubator/kubecf/pull/1162
